src/misc/lv_style.c

```diff
@@ -67,11 +166,36 @@ void lv_style_reset(lv_style_t * style)
 #endif
 }
 
-lv_style_prop_t lv_style_register_prop(void)
+lv_style_prop_t lv_style_register_prop(uint8_t flag)
+{
+    /*
+     * Allocate the lookup table if it's not yet available.
+     */
+    if(lv_style_custom_prop_flag_lookup_table == NULL) {
+        lv_style_custom_prop_flag_lookup_table = lv_mem_alloc(LV_STYLE_MAX_CUSTOM_PROPS);
```

Comment:
Why not `realloc` instead of using a fixed `LV_STYLE_MAX_CUSTOM_PROPS` size?

Reply:
Even with 256 custom properties, this would only be 256 bytes (and the default is 64). In my opinion it will just add a greater risk of fragmentation for not much gain. We allocate a lot more than 256 bytes in other places.

Reply:
I see 2 advantages of `reallocing`:
- There is no max prop. count -> one less issue, and apps can scale better 
- One less config option

If fragmentation is a concern what about realloc-ing in 16 or 32 byte blocks?

Reply:
> There is no max prop. count -> one less issue, and apps can scale better

Fair enough, I'll implement it that way, then.

Reply:
Implemented.